docs/guides/basic-rbac.mdx

```diff
@@ -3,267 +3,303 @@ title: Implement basic Role Based Access Control (RBAC) with metadata
+  }
+  ```
 
-This `checkRole()` helper will accept a role using the `Roles` type and will return `true` if the user has that role, or `false` if the user does not.
+  ### Add admin tools to find users and manage roles
 
-Next, the admin dashboard can be refactored to use the `checkRole()` helper. Navigate back to the admin dashboard file. In the `if()` statement, remove the check that was used previously and replace it with the new `checkRole()` helper with `"admin"` as the argument.
+  You can use the `checkRole()` function along with server actions to build basic tools for finding users and managing roles.
 
-```tsx {{ filename: 'app/admin/dashboard/page.tsx', mark: [2, [5, 8]] }}
-import { redirect } from 'next/navigation'
-import { checkRole } from '@/utils/roles'
+  Create a server action for managing a user's role.
 
-export default function AdminDashboard() {
-  // If the user does not have the admin role, redirect them to the home page
-  if (!checkRole('admin')) {
-    redirect('/')
-  }
+  1. In your `app/admin/` directory, create an `_actions.ts` file.
+  1. Create a server action that uses the `checkRole()` function to verify that the current user has the `admin` role. If they do, proceed to update the specified user's role using Clerk's [JavaScript Backend SDK](/docs/references/backend/user/update-user). This ensures that only administrators can modify user roles.
 
-  return (
-    <>
-      <h1>This is the admin dashboard</h1>
-      <p>This page is restricted to users with the 'admin' role.</p>
-    </>
-  )
-}
-```
+  ```ts {{ filename: 'app/admin/_actions.ts' }}
+  'use server'
 
-> [!NOTE]
-> You can modify the behavior of the helper function to meet your needs. Maybe it will return the roles that the user has, or you could create a `protectByRole()` and have that function handle the redirect.
+  import { checkRole } from '@/utils/roles'
+  import { clerkClient } from '@clerk/nextjs/server'
 
-## Add admin tools to find users and add roles
+  export async function setRole(formData: FormData) {
+    // Check that the user trying to set the role is an admin
+    if (!checkRole('admin')) {
+      return { message: 'Not Authorized' }
+    }
 
-You can leverage the `checkRole()` function you added along with server actions to build basic tools for finding users and managing roles.
+    try {
+      const res = await clerkClient().users.updateUser(formData.get('id') as string, {
+        publicMetadata: { role: formData.get('role') },
+      })
+      return { message: res.publicMetadata }
+    } catch (err) {
+      return { message: err }
+    }
+  }
+  ```
+
+  With the server action set up, now build the `<SearchUsers />` component. This component includes a form for searching users, and when submitted, appends the search term to the URL as a search parameter. Your `/admin` route will then perform a query based on the updated URL.
+
+  1. In your `app/admin/` directory, create a `_search-users.tsx` file.
```

Comment:
Not required, but I would either:
1. Use the name `_components.tsx` as that's a little more common
2. Move the file and use `/components/SearchUsers.tsx` as that's definitely more common.